hyperspy/axes.py

```diff
@@ -655,6 +720,18 @@ def _append_axis(self, *args, **kwargs):
         axis.axes_manager = self
         self._axes.append(axis)
 
+    def _on_index_changed(self):
+        self._update_attributes()
+        self.events.indices_changed.trigger(axes_manager=self)
+
+    def _on_view_changed(self):
+        self._update_attributes()
+        self.events.axes_changed.trigger()
```

Comment:
I agree that this is not the time and place to come out with all the possibly useful events. However, here you have found one that will be doubtlessly useful. I don't see why we shouldn't create it at this point.